holmes/core/supabase_dal.py

```diff
@@ -1,52 +1,108 @@
+import base64
+import json
 import logging
 import os
-from datetime import datetime
-from typing import Dict, Optional
+from typing import Dict, Optional, List
 
+import yaml
 from supabase import create_client
 from supabase.lib.client_options import ClientOptions
 from pydantic import BaseModel
 
+from holmes.common.env_vars import (ROBUSTA_CONFIG_PATH, ROBUSTA_ACCOUNT_ID, STORE_URL, STORE_API_KEY, STORE_EMAIL,
+                                    STORE_PASSWORD)
+
+
 SUPABASE_TIMEOUT_SECONDS = int(os.getenv("SUPABASE_TIMEOUT_SECONDS", 3600))
 
 ISSUES_TABLE = "Issues"
 EVIDENCE_TABLE = "Evidence"
 
 
+class RobustaConfig(BaseModel):
+    sinks_config: List[Dict[str, Dict]]
+
+
+class RobustaToken(BaseModel):
+    store_url: str
+    api_key: str
+    account_id: str
+    email: str
+    password: str
+
+
 class SupabaseDal:
-    def __init__(
-        self,
-        url: str,
-        key: str,
-        email: str,
-        password: str,
-    ):
-        self.url = url
-        self.key = key
+
+    def __init__(self):
+        self.enabled = self.__init_config()
+        if not self.enabled:
+            logging.info("Robusta store initialization parameters not provided. skipping")
+            return
+        logging.info(f"Initializing robusta store for account {self.account_id}")
         options = ClientOptions(postgrest_client_timeout=SUPABASE_TIMEOUT_SECONDS)
-        self.client = create_client(url, key, options)
-        self.email = email
-        self.password = password
+        self.client = create_client(self.url, self.api_key, options)
         self.sign_in()
 
+    @staticmethod
+    def __load_robusta_config() -> Optional[RobustaToken]:
+        config_file_path = ROBUSTA_CONFIG_PATH
+        if not os.path.exists(config_file_path):
+            logging.info(f"No robusta config in {config_file_path}")
+            return None
+
+        logging.info(f"loading config {config_file_path}")
+        with open(config_file_path) as file:
+            yaml_content = yaml.safe_load(file)
+            config = RobustaConfig(**yaml_content)
+            for conf in config.sinks_config:
+                if "robusta_sink" in conf.keys():
+                    token = conf["robusta_sink"].get("token")
+                    return RobustaToken(**json.loads(base64.b64decode(token)))
+
+        return None
+
+    def __init_config(self) -> bool:
+        robusta_token = self.__load_robusta_config()
```

Comment:
Could you document the behaviour somewhere - either in the helm values or in this file - that we load from robusta token on disk first if possible, and if not fallback to environment variables if given

Reply:
I'll put a comment here